app.rb

```diff
@@ -6,16 +6,54 @@
 $polls = [
   Poll.new('好きな料理', ['肉じゃが', 'しょうが焼き', 'から揚げ']),
   Poll.new('人気投票', ['おむすびけん', 'クックパッドたん']),
+  Poll.new('期限を過ぎた投票', ['a', 'b'], Time.now)
 ]
 
 get '/' do
   erb :index, locals: { polls: $polls }
 end
 
+get '/polls/new' do
+  erb :new_poll
+end
+
+post '/polls/new' do
+  halt 404, '選択肢を2つ以上入力してください' if params['candidates'].length < 1
```

Comment:
404 はリソースが見つからない場合のステータスコードなので、ユーザーの入力値が不正な場合は [400 Bad Request](https://developer.mozilla.org/ja/docs/Web/HTTP/Status/400) あたりを返すのが適切かなと思います。

```suggestion
  halt 400, '選択肢を2つ以上入力してください' if params['candidates'].length < 1
```